command/agent/csi_endpoint.go

```diff
@@ -116,3 +114,56 @@ func (s *HTTPServer) csiVolumeDelete(id string, resp http.ResponseWriter, req *h
 
 	return nil, nil
 }
+
+// CSIPluginsRequest lists CSI plugins
+func (s *HTTPServer) CSIPluginsRequest(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	if req.Method != "GET" {
+		return nil, CodedError(405, ErrInvalidMethod)
+	}
+
+	args := structs.CSIPluginListRequest{}
+
+	if s.parse(resp, req, &args.Region, &args.QueryOptions) {
```

Comment:
I hadn't seen [`parseWait`](https://github.com/hashicorp/nomad/blob/f4f911cfd80776e25245b679f25fa2a3e9715c05/command/agent/http.go#L404) before, so this threw me for a loop.  If parsing failed why don't we return a 400 error? (It looks like we always `return nil, nil` on other endpoints too, so it's typical across the code base. Just wanted to understand the reasoning.)